test/lib/logger.test.js

```diff
@@ -0,0 +1,169 @@
+const chai = require('chai')
+const spies = require('chai-spies')
+const proxyquire = require('proxyquire')
+const path = require('path')
+
+const expect = chai.expect
+chai.use(spies)
+
+const CORRELATION_ID = 'CORRELATION_ID'
+const MODULE_NAME = path.basename(path.resolve())
+const MOCK_DATE = 1558625994627
+
+const LOG_LEVELS = require('../../lib/log-levels')
+
+const { Logger } = proxyquire('../../lib/logger', {
+  '@first-lego-league/ms-correlation': {
+    getCorrelationId () {
+      return CORRELATION_ID
+    }
+  }
+})
+
+describe('Logger', () => {
+  it('returns a logger even if not called as a constructor', () => {
+    expect(Logger() instanceof Logger).to.equal(true)
+  })
+
+  it('has a default level of DEBUG', () => {
+    expect(Logger().logLevel).to.equal(LOG_LEVELS.DEBUG)
+  })
+
+  describe('set log level', () => {
+    it('sets the log level if its valid', () => {
+      const logger = Logger()
+      logger.setLogLevel(LOG_LEVELS.INFO)
+      expect(logger.logLevel).to.equal(LOG_LEVELS.INFO)
+    })
+
+    it('throws an error if the log level is not one of the allowed log levels', () => {
+      const logger = Logger()
+      expect(() => logger.setLogLevel(-5)).to.throw()
+    })
+  })
+
+  describe('log', () => {
+    const sandbox = chai.spy.sandbox()
+    const originalDateNow = Date.now
+
+    beforeEach(() => {
+      sandbox.on(console, ['log'])
+      Date.now = () => MOCK_DATE
+    })
+
+    afterEach(() => {
+      sandbox.restore()
+      Date.now = originalDateNow
+    })
+
+    it('does not log if the level is lower then the logger\'s level', () => {
+      const logger = Logger()
+      logger.setLogLevel(LOG_LEVELS.WARN)
+      logger.log(LOG_LEVELS.INFO, 'Some Log')
+      expect(console.log).to.not.have.been.called()
+    })
+
+    it('logs if the level is equal to the logger\'s level', () => {
+      const logger = Logger()
+      logger.setLogLevel(LOG_LEVELS.WARN)
+      logger.log(LOG_LEVELS.WARN, 'Some Log')
+      expect(console.log).to.have.been.called.exactly(1)
+    })
+
+    it('logs if the level is higher then the logger\'s level', () => {
+      const logger = Logger()
+      logger.setLogLevel(LOG_LEVELS.WARN)
+      logger.log(LOG_LEVELS.FATAL, 'Some Log')
+      expect(console.log).to.have.been.called.exactly(1)
+    })
+
+    it('logs the formatter log according the logger\'s value formatting method', () => {
+      const logger = Logger()
+      logger.formatLog = chai.spy(() => { })
```

Comment:
It's also seems like the function `formatLog` should be protected by and be called `_formatLog`

Reply:
I think so too, but again, I didn't want to change the design of the package.

Reply:
ok